src/Http/Authentication.Abstractions/src/AuthenticationTicket.cs

```diff
@@ -52,5 +52,13 @@ public AuthenticationTicket(ClaimsPrincipal principal, string authenticationSche
         /// Additional state values for the authentication session.
         /// </summary>
         public AuthenticationProperties Properties { get; private set; }
+
+        /// <summary>
+        /// Returns a copy of the ticket
+        /// </summary>
+        /// <returns>A copy of the ticket</returns>
+        public AuthenticationTicket Clone()
+            => new AuthenticationTicket(new ClaimsPrincipal(Principal), new AuthenticationProperties(Properties.Items, Properties.Parameters), AuthenticationScheme);
```

Comment:
These are still shallow clones for Principal and Properties. It's still the same mutable Identity instances and property collections.

Reply:
I added a deeper clone for properties and used Principal.Clone which still results in cloning the identities, is it reasonable for us to just iterate over each identity in the principal and create a new claimsprincipal and basically do a 'claims' snapshot clone of the principal for the cache?